src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const changesOfState = [];
+  const clone = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(clone, action.extraData);
+      changesOfState.push({ ...clone });
```

Comment:
Don't repeat yourself.
You can wright `changesOfState.push({ ...clone });` only once (in the end of the iterarion)

Reply:
Done!